src/daft-logical-plan/src/builder/resolve_expr.rs

```diff
@@ -151,6 +151,80 @@ fn col_resolves_to_plan(column: &UnresolvedColumn, plan: &LogicalPlanRef) -> Daf
     })
 }
 
+fn replace_element_with_column_ref(expr: ExprRef, replacement: ExprRef) -> DaftResult<ExprRef> {
+    expr.transform(|e| {
+        if matches!(
+            e.as_ref(),
+            Expr::Column(Column::Unresolved(UnresolvedColumn {
+                name,
+                plan_ref: PlanRef::Unqualified,
+                plan_schema: None
+            })) if name.as_ref() == ""
+        ) {
+            Ok(Transformed::yes(replacement.clone()))
+        } else {
+            Ok(Transformed::no(e))
+        }
+    })
+    .map(|res| res.data)
+}
+
+fn resolve_list_evals(expr: ExprRef) -> DaftResult<ExprRef> {
+    if expr.has_element() {
+        expr.transform_down(|e| {
+            let expr_ref = e.as_ref();
+            if let Expr::ScalarFunction(sf) = expr_ref {
+                // the `list` type should always be the first element
+                let inputs = sf.inputs.clone();
+                let list_col = inputs
+                    .first()
+                    .expect("list should have at least one element");
+
+                let exploded = list_col.clone().explode()?;
+
+                let mut new_inputs = Vec::with_capacity(inputs.len());
+                new_inputs.push(FunctionArg::Unnamed(list_col.clone()));
+
+                for input in inputs.iter().skip(1) {
```

Comment:
there should only be two inputs right? Maybe we can somehow use the FunctionArgs derive macro here to make the input arguments easier to parse.